src/AzureRMAssets.ts

```diff
@@ -116,26 +100,63 @@ export interface VersionRedirect {
 }
 
 /**
- * Metadata for a TLE function.
+ * Metadata for all TLE (Template Language Expression) functions.
+ */
+export class FunctionsMetadata {
+    public constructor(public readonly functionMetadata: FunctionMetadata[]) {
+    }
+
+    public findbyName(functionName: string): FunctionMetadata | undefined {
+        const lowerCasedFunctionName: string = functionName.toLowerCase();
+        for (const functionMetadata of this.functionMetadata) {
+            if (functionMetadata.lowerCaseName === lowerCasedFunctionName) {
+                return functionMetadata;
+            }
+        }
+
```

Comment:
Given that `this.functionMetadata` is an array (line 106),  isn't this equivalent to `this.functionMetadata.find(functionMetadata => functionMetadata.lowerCaseName === lowerCasedFunctionName)` ?

Reply:
Yes. I had actually had that at some point, seem to have lost it.